### PR TITLE
Implement UpdateGroup

### PR DESCRIPTION
**Pull request by bduffany**

### CODE REVIEW

File: server/buildbuddy_server/buildbuddy_server.go [hunk 1]
```diff
@@ -239,6 +239,38 @@ func (s *BuildBuddyServer) CreateGroup(ctx context.Context, req *grpb.CreateGrou
...
+	if err := s.authorizeGroupAccess(ctx, req.GetId()); err != nil {
+		return nil, err
+	}
+	group := &tables.Group{
+		GroupID: req.GetId(),
+	}
+	if err := userDB.FillGroup(ctx, group); err != nil {
+		return nil, err
+	}
+	group.Name = req.GetName()
```
Comment (tylerwilliams) [inline server/buildbuddy_server/buildbuddy_server.go:257]: can this live above when you define the group?

group := &tables.Group{
                Name: req.GetName(),
		GroupID: req.GetId(),
	}

**Review by tylerwilliams — CHANGES_REQUESTED** [review]

File: server/buildbuddy_server/buildbuddy_server.go (hunk 1 above)
Comment (bduffany) [reply server/buildbuddy_server/buildbuddy_server.go:257]: I had the same thought, but realized `FillGroup` would clobber the name. Yet another reason to get rid of `FillGroup`

Comment (tylerwilliams) [reply server/buildbuddy_server/buildbuddy_server.go:257]: yuck! can't wait to be rid of that thing.

**Review by tylerwilliams — APPROVED** [review]
LGTM

**Pull request merged**